### PR TITLE
chore(EMS-3643-3953): eligibility - refactor companies house input hint

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:
This PR fixes some tech debt where the Companies House search input hint had:
  - Hard coded HTML in the content strings.
  - Missing E2E test coverage.

## Resolution :heavy_check_mark:
- Update the content strings to be pure content strings.
- Update nunjucks template.
- Add E2E test.

## Miscellaneous :heavy_plus_sign:
- Address TODO comment RE asserting a status after "save and back" functionality (EMS-3828).

### CODE REVIEW

File: e2e-tests/commands/insurance/complete-how-much-the-agent-is-charging-form.js [hunk 1]
```diff
@@ -12,7 +12,9 @@ const {
  * @param {String} fixedSumAmount: Fixed sum amount
  */
 const completeHowMuchTheAgentIsChargingForm = ({ fixedSumAmount = application.EXPORT_CONTRACT.AGENT_CHARGES[FIXED_SUM_AMOUNT] }) => {
-  cy.keyboardInput(field(FIXED_SUM_AMOUNT).input(), fixedSumAmount);
+  if (fixedSumAmount) {
```
Comment (Zainzzkk) [inline e2e-tests/commands/insurance/complete-how-much-the-agent-is-charging-form.js:15]: do we need the if statement if this function doesn't do anything else? 
Maybe completeHowMuchTheAgentIsChargingForm should only be called if fixedSumAmount

Comment (ttbarnes) [reply e2e-tests/commands/insurance/complete-how-much-the-agent-is-charging-form.js:15]: Good shout - this was for submitting a form via "Save and back" without entering anything. I've simplified the test and removed the condition from the command.

**Review by Zainzzkk — APPROVED** [review]

File: src/ui/templates/insurance/eligibility/companies-house-search.njk [hunk 2]
```diff
@@ -40,9 +40,13 @@
...
         <div class="companies-house-search">
 
+          {% set hintHtml %}
+            <p><span data-cy="{{ FIELD.ID }}-hint-intro">{{ FIELD.HINT.INTRO }} </span><a class="govuk-link" href="{{ FIELD.HINT.LINK.HREF }}" data-cy="{{ FIELD.ID }}-hint-link">{{ FIELD.HINT.LINK.TEXT }}</a>.</p>
+          {% endset %}
+
           {{ textInput.render({
             id: FIELD.ID,
-            hintHtml: FIELD.HINT,
+            hintHtml: hintHtml,
```
Comment (abhi-markan) [inline src/ui/templates/insurance/eligibility/companies-house-search.njk:49]: Can it be `hintHtml` only?

Comment (ttbarnes) [reply src/ui/templates/insurance/eligibility/companies-house-search.njk:49]: Unfortunately not, Nunjucks

**Pull request merged**